tests/tracking/test_rest_tracking.py

```diff
@@ -4666,3 +4666,75 @@ def test_get_provider_config(mlflow_client_with_secrets):
     # Missing provider parameter returns 400
     response = requests.get(f"{base_url}/ajax-api/3.0/mlflow/endpoints/provider-config")
     assert response.status_code == 400
+
+
+def test_endpoint_with_orphaned_model_definition(mlflow_client_with_secrets):
+    store = mlflow_client_with_secrets._tracking_client.store
+
+    secret = store.create_gateway_secret(
+        secret_name="orphan-test-key",
+        secret_value={"api_key": "sk-orphan-test"},
+        provider="openai",
+    )
+
+    model_def = store.create_gateway_model_definition(
+        name="orphan-model-def",
+        secret_id=secret.secret_id,
+        provider="openai",
+        model_name="gpt-4",
+    )
+
+    endpoint = store.create_gateway_endpoint(
+        name="orphan-test-endpoint",
+        model_definition_ids=[model_def.model_definition_id],
+    )
+
+    assert len(endpoint.model_mappings) == 1
+    assert endpoint.model_mappings[0].model_definition.secret_id == secret.secret_id
+    assert endpoint.model_mappings[0].model_definition.secret_name == "orphan-test-key"
+
+    store.delete_gateway_secret(secret.secret_id)
+
+    fetched_endpoint = store.get_gateway_endpoint(endpoint.endpoint_id)
+    assert len(fetched_endpoint.model_mappings) == 1
+    assert fetched_endpoint.model_mappings[0].model_definition.secret_id is None
+    assert fetched_endpoint.model_mappings[0].model_definition.secret_name is None
+
+    store.delete_gateway_endpoint(endpoint.endpoint_id)
```

Comment:
Is this for data clean-up? Then I think that is unnecessary since db is reset in the `mlflow_client_with_secrets` fixture